src/System.Data.Common/src/System/Data/Common/DbDataReader.cs

```diff
@@ -56,7 +56,7 @@ protected virtual void Dispose(bool disposing)
 
         public virtual DataTable GetSchemaTable()
         {
-            throw new NotSupportedException();
+            throw ADP.MethodNotImplemented();
```

Comment:
I am not in favor of changing the exception type from the base class.

Reply:
It still throws a NotImplementedException.

Reply:
But we were throwing a `NotSupportedException`

Reply:
Missed that. I'll just remove these other library changes.

Reply:
I understand that changing the exception type might be considered a breaking change, but I think *not implemented* is exactly what should be thrown here. That would inform those deriving from this class that they are expected to *implement* this method, whereas the current exception is misleading in that it gives the impression that this functionality is not supported (and provides no indication or incentive to implement).

Reply:
That's maybe a separate issue though. Thank you @corivera for improving the confusing stacktrace in `TdsExecuteRPC`.

Reply:
@nil4 I agree that ADO.Net should have thrown a NotImplemented from the very beginning and you have rightly recognized that changing the exception type would be a breaking change.   We need to stay compatible with .Net Framework.
Had you discovered where the exception was coming from, it would have been easier for you to develop your customer DBDataReader. So though we won't change the exception type, surfacing the full stack trace would definitely alleviate the issue to a great extent.